paimon-core/src/main/java/org/apache/paimon/schema/SchemaManager.java

```diff
@@ -283,7 +283,7 @@ public TableSchema commitChanges(List<SchemaChange> changes) throws Exception {
                             update.fieldName(),
                             (field) -> {
                                 checkState(
-                                        DataTypeCasts.supportsImplicitCast(
```

Comment:
By allowing explicit type casting you will allow casting from `string` to `bigint`, or `timestamp` to `tinyint`, etc., etc. Have you implemented all these in `CastExecutors`?

Reply:
This PR is to quickly solve the known BUG of users, support bigint conversion string, I have created a new issue to supports all explicit casting.